Bottom: 2c5ef05bd4b4b7144d2adeebce08ba9fa5b6b3dd
Top:    0d7c0d90d4220d64f75161279b13594bbca7f93d
Author: Peter Clifton <peter@clifton-electronics.co.uk>
Date:   2016-01-07 02:18:22 +0000

polygon.c: For some reason, the range computation in frac_circle appears to be wrong


---

diff --git a/src/polygon.c b/src/polygon.c
index acd55c5..91a16a8 100644
--- a/src/polygon.c
+++ b/src/polygon.c
@@ -483,8 +483,8 @@ frac_circle (PLINE * c, Coord X, Coord Y, Vector v, int fraction)
   e1 = (v[0] - X) * POLY_CIRC_RADIUS_ADJ;
   e2 = (v[1] - Y) * POLY_CIRC_RADIUS_ADJ;
 
-  /* NB: the caller adds the last vertex, hence the -1 */
-  range = POLY_CIRC_SEGS / fraction - 1;
+  /* XXX */ /* NB: the caller adds the last vertex, hence the -1 */
+  range = POLY_CIRC_SEGS / fraction;
   for (i = 0; i < range; i++)
     {
       /* rotate the vector */
